base/reflection.jl

```diff
@@ -179,6 +194,10 @@ function methods(f::ANY,t::ANY)
     Any[m[3] for m in _methods(f,t,-1)]
 end
 function _methods(f::ANY,t::ANY,lim)
+    #special case Expr to allow methods(@time,Tuple{Any})
+    if _is_empty_macrocall(f)
+        f = _get_macro(current_module(),f.args[1])
```

Comment:
This code is incorrect --- it makes it impossible for the system to query the methods of an `Expr` object. Exprs don't have any methods of course, but that just means this function must return an empty array.